src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java

```diff
@@ -359,15 +403,44 @@ <U> boolean update(LockedContext<U> context,
                        Function<T, T> updater,
                        BooleanSupplier updateNext) {
         final RelationalDaoPriv dao = daos.get(context.getShardId());
-
+        final ScrollParamPriv<T> scrollParam = ScrollParamPriv.<T>builder()
+                .criteria(criteria)
+                .build();
         try {
-            final ScrollParamPriv<T> scrollParam = ScrollParamPriv.<T>builder()
-                    .criteria(criteria)
-                    .build();
-
+            return transactionExecutor.<ScrollableResults, ScrollParamPriv<T>, Boolean>execute(context.getSessionFactory(),
```

Comment:
Try got new formatted transactionExecutor but exception code is replaced by original transactionExecutor instead of  throw new RuntimeException("Error updating entity with scroll: " + criteria, e);